src/argilla_server/models/database.py

```diff
@@ -352,6 +352,11 @@ def is_draft(self):
     def is_ready(self):
         return self.status == DatasetStatus.ready
 
+    def metadata_property_by_name(self, name: str) -> Union["MetadataProperty", None]:
+        for metadata_property in self.metadata_properties:
+            if metadata_property.name == name:
+                return metadata_property
```

Comment:
Not sure about this one. I guess SQLAlchemy should have some way to filter by a relationship collection.

Reply:
Yes, but, since all relationships are loaded when adding records (with_questions=True, with_fields=True...), I was thinking is easier doing this than querying the DB